### ISSUE TITLE
Can't find `/usr/lib/libshaderc_shared.so`

### ISSUE THREAD

**Issue by Rua** [google/shaderc-rs#107]
I downloaded the pre-built shaderc library package for Linux, and unpacked it in `/usr/lib`. This is one of the locations where shaderc-rs will search for libraries according to the readme, and `/usr/lib/libshaderc_shared.so` is right there along with all the other shaderc libraries. Yet for some reason it fails to find the library, and proceeds to build from source (which fails because I uninstalled cmake). How come it's not finding the library? Is the documentation incorrect?

**Comment by antiagainst** [comment]
Hi, note that `/usr/lib/` is just one example given for Linux system library paths: "On Linux, system library paths like ..". There is an implicit ordering in the library path: 

https://github.com/google/shaderc-rs/blob/21bb994197b536acabd1ddfd3ca82f27d1150645/shaderc-sys/build/build.rs#L180-L191

And `/usr/lib/` is the last one of it. I think the problem is that only directory existence was checked there; not the library. So as long as you have other directories recognized, they will be picked by the script. 

This is all very tricky and ad-hoc. I think the above part should be rewritten to just use some more standardized way to probe system shared libraries.

Hopefully this clarifies.

**Comment by nicoonoclaste** [comment]
> This is all very tricky and ad-hoc. I think the above part should be rewritten to just use some more standardized way to probe system shared libraries.

Yes, using `pkg-config` would be the standard way to solve the problem on (AFAIK) all opensource distros, be it Linux, *BSD, etc.

As it stands, this prevents using prebuilt versions of `shaderc` on NixOS (or any dev environment managed with nix) as they would be found under `/nix/store/{hash}-shaderc-{version}-{dev, lib}/`: pkg-config would find them through its environment variables, which are automatically set by a nix setup-hook for pkg-config.

PS: I was able to work around that, by setting `SHADERC_LIB_DIR = lib.makeLibraryPath [ pkgs.shaderc ];`.  It would still make sense to use `pkg-config`.

**Comment by antiagainst** [comment]
#160 added support to query from `pkg-config`, which hopefully addresses the issue. Please double check.
